src/TaskCreationSection/CategoryCreationSection/AddCategoryModal/IconPicker.tsx

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable react/display-name */
+/* eslint-disable react-hooks/exhaustive-deps */
+/* eslint-disable react/require-default-props */
+
+import { memo, useMemo } from 'react'
+import { IconContext } from '@react-icons/all-files'
+import { IoHappy } from '@react-icons/all-files/io5/IoHappy'
+import iconList from '../../../Helpers/iconList'
+
+type IconPickerProps = {
+    newCategoryIcon: string
+    setNewCategoryIcon: (categoryIcon: string) => void
+}
+const IconPicker = memo(
+    ({ newCategoryIcon, setNewCategoryIcon }: IconPickerProps): JSX.Element => {
+        const iconContextValue = useMemo(
+            () => ({
+                size: '3em',
+                display: 'flex',
+                color: '#334155',
+            }),
+            []
+        )
+        return (
+            <IconContext.Provider value={iconContextValue}>
```

Comment:
I'd extract this to separate file as we already did for another Context, let's talk about abstraction layer pros on the call